src/main/java/org/opensrp/web/rest/LocationResource.java

```diff
@@ -411,6 +424,31 @@ public void setLocationService(PhysicalLocationService locationService) {
 
 	}
 
+	/**
+	 * This methods provides an API endpoint that searches for all location ids
+	 * sorted by server  version ascending
+	 *
+	 * @return A list of location Ids and last server version
+	 */
+	@RequestMapping(value = "/findLocationIds", method = RequestMethod.GET, produces = {
+			MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<Identifier> findLocationIds(
+			@RequestParam(value = SERVER_VERSION)  long serverVersion) {
+
+		try {
+			Pair<List<String>, Long> locationIdsPair = locationService.findAllLocationIds(serverVersion, DEFAULT_GET_ALL_IDS_LIMIT);
+			Identifier identifiers = new Identifier();
+			identifiers.setIdentifiers(locationIdsPair.getLeft());
+			identifiers.setLastServerVersion(locationIdsPair.getRight());
+			
+			return new ResponseEntity<>(identifiers, RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
+		} catch (Exception e) {
```

Comment:
can we catch something more specific?

Reply:
This catches all possible errors preventing leaking of stack-trace. This can be removed the global error handler is configured https://github.com/OpenSRP/opensrp-server-web/issues/245

Reply:
OK, how about adding inline TODOs to all of these, and also post merge links to the master code-lines to #245 so it's easy to come back and remove these

Reply:
done